### ISSUE TITLE
Subscriptions

### ISSUE THREAD

**Issue by jmeas** [jmeas/react-request#65]
React Apollo's HoC subscribes you to updates for the same query.

This probably doesn't make sense for this component, but where do I draw the line between what this component does and what the next level should do? As a comparison, ApolloLink, the lib that is roughly analogous to this in Apollo, doesn't do some of the stuff that this does.

**Comment by jmeas** [comment]
Apollo Link: deduplication + HTTP.
Apollo Client: cache layer + API to interact with Apollo Link
React Apollo: React sugar on top of Apollo Client functionality

---

I think the current direction this lib is going in is fine, in relation to Apollo's stack.

**Comment by jmeas** [comment]
I need some sort of API to sort updating when another request updates the cache. Here's what I mean:

FetchA mounts and gets `/books/2`.

10 minutes later, FetchB mounts and gets the same data, but it has changed. The `FetchA` component should receive an update.

---

To do this, I will need to do the following:

1. a request ending does not wipe the cache. Instead, it moves it to an "active" queue.
2. there will be an API to remove a cached request from the queue. This will be done when the `<Fetch/>` is unmounted.

---

**Alternatively** – there may be a way to add this in a wrapping component. I'll need to think more about that...

**Comment by jmeas** [comment]
Above, I proposed the following idea:

> To do this, I will need to do the following:

> 1. a request ending does not wipe the cache. Instead, it moves it to an "active" queue.
> 2. there will be an API to remove a cached request from the queue. This will be done when the <Fetch/> is unmounted.

this idea does not work because `fetch-dedupe` works with Promises, and Promises can only be `resolved` one time. So after the first request is made, I have no handlers to the component to continue pushing updates.

tl;dr:

fetch-dedupe _cannot_ be used for this system, but the component _could_ manage a cache for this.
